tests/cases.py

```diff
@@ -0,0 +1,98 @@
+# -*- coding: utf-8 -*-
+
+"""Test cases for PyKEEN."""
+
+import tempfile
+import timeit
+import unittest
+from typing import ClassVar, Optional, Type
+
+from requests.exceptions import ConnectionError
+
+from pykeen.datasets.base import LazyDataSet
+from pykeen.triples import TriplesFactory
+
+
+class DatasetTestCase(unittest.TestCase):
+    """A test case for quickly defining common tests for datasets."""
+
+    #: The expected number of entities
+    exp_num_entities: ClassVar[int]
+    #: The expected number of relations
+    exp_num_relations: ClassVar[int]
+    #: The dataset to test
+    dataset_cls: ClassVar[Type[LazyDataSet]]
+    #: The instantiated dataset
+    dataset: LazyDataSet
+
+    #: Should the validation be assumed to have been loaded with train/test?
+    autoloaded_validation: ClassVar[bool] = False
+
+    def test_dataset(self):
+        """Generic test for datasets."""
+        self.assertIsInstance(self.dataset, LazyDataSet)
+
+        # Not loaded
+        self.assertIsNone(self.dataset._training)
+        self.assertIsNone(self.dataset._testing)
+        self.assertIsNone(self.dataset._validation)
+        self.assertFalse(self.dataset._loaded)
+        self.assertFalse(self.dataset._loaded_validation)
+
+        # Load
+        try:
+            self.dataset._load()
```

Comment:
This will download the dataset for every build, correct?

Reply:
... yes. we discussed this briefly in a recent PR when i added the try/except for failures. Now that I've got the idea for writing mocks, next PR will include mocks for the other base classes and won't actually ask for the live datasets